### PR TITLE
amend UVIP KEP with some additional details

### PR DESCRIPTION
**Pull request by logicalhan**

### CODE REVIEW

File: keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md [hunk 1]
```diff
@@ -190,19 +190,39 @@ incorrectly or objects being garbage collected mistakenly.
...
-* A controller adds the apiserver to the list of alternates for its built-in
+* A controller adds the apiserver to the list of `serviceableBy`s for its built-in
   group-versions.
 * The same controller removes expired apiservers from the list. (Enabled by the
   apiserver identity work.)
 * A new handler is added to the stack:
   - If the request is for a group/version the apiserver doesn't have locally, it
-    will proxy the request to one of the alternates instead.
+    will proxy the request to ag `serviceableBy`s if one is available, if one is not 
+    available,then we will return a 503 (there is a possibility of a race
```
Comment (lavalamp) [inline keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md]:
```suggestion
    available, then we will return a 503 (there is a possibility of a race 
```

File: keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md [hunk 2]
```diff
@@ -258,7 +278,12 @@ TODO: security / cert stuff.
 
 ## Design Details
 
-TODO: specific API change (x2)
+We need two API changes:
+
+1. The apiservices API needs to be augmented with a `serviceableBy` field, which
```
Comment (lavalamp) [inline keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md]: I'm unsure we need this now! StorageVersion gives us apiserver id choices for a GVR, no?

Comment (logicalhan) [reply keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md]: deleted

File: keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md [hunk 3]
```diff
@@ -189,19 +189,34 @@ incorrectly or objects being garbage collected mistakenly.
...
-  apiservers which believe they can serve the group-version.
+We will use the existing `StorageVersion` API to figure out which group, versions,
+and resources an apiserver can serve.
+
+We will also need to make discover reports the same set of resources everywhere. We propose
+routing discovery requests from old-apiservers to the new api-server, so that all discovery
+requests reflect the newest one. We specifically rule out merging discovery docs, because
+merging discovery is:
+
+* complicated
```
Comment (lavalamp) [inline keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md]: this part still seems wrong to me or at least needs to go in an unresolved block

Comment (logicalhan) [reply keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md]: Sure moved.

File: keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md [hunk 4]
```diff
@@ -189,19 +189,20 @@ incorrectly or objects being garbage collected mistakenly.
...
 * A new handler is added to the stack:
-  - If the request is for a group/version the apiserver doesn't have locally (we
-    can use the StorageVersion API), it will proxy the request to one of the
-    alternates instead.
+
+  - If the request is for a group/version/resource the apiserver doesn't have
+    locally (we can use the StorageVersion API), it will proxy the request to
+    one of the apiservers that is listed in the object. If an apiserver fails
+    to respond is not available, then we will return a 503 (there is a small
+    possibility of a race between the controller registering the apiserver
```
Comment (lavalamp) [inline keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md:203]: I'm gonna merge this but we need a better story for this race, IMO

**Pull request merged**